TROUBLESHOOTING.md

```diff
@@ -0,0 +1,163 @@
+the `ruby-lsp` in their `Gemfile`, we currently create a custom bundle under the `.ruby-lsp` directory inside your
+project. That directory contains another `Gemfile`, that includes the `ruby-lsp` gem in addition to your project's
+dependencies. This approach allows us to automatically detect which formatter your project uses and which gems we need
+to index for features such as go to definition.
+
+**Note**: we are working with the rubygems/bundler team to have this type of mechanism properly supported from within
+Bundler itself, which is currently being experimented with in a plugin called `bundler-compose`. Once `bundler-compose`
+is production ready, the entire custom bundle created under the `.ruby-lsp` directory will go away and we'll rely on
+Bundler to compose the LOAD_PATH including the `ruby-lsp` gem.
+
+## Common issues
+
+There are two main sources of issues users typically face during activation: shell or Bundler related problems.
+
+### Shell issues
+
+When the extension invokes the shell and loads its config file (`~/.zshrc`, `~/.bashrc`, etc), it is susceptible to
+issues that may be caused by how the shell or its plugins interact with the NodeJS process. For example
+
+- Some plugins completely redirect the stderr pipe to implement their functionality (fixed on the Ruby LSP side by
+  https://github.com/Shopify/vscode-ruby-lsp/pull/918)
+- Some plugins fail immediately or end up in an endless loop if they detect there's no UI attached to the shell process.
+  In this case, it's not possible to fix from the Ruby LSP side since a shell invoked by NodeJS will never have a UI
+
+Additionally, some users experience an issue where VS Code selects the wrong shell, not respecting the `SHELL`
+environment variable. This usually ends up in having `/bin/sh` selected instead of your actual shell. If you are facing
+this problem, please try to
+
+- Update VS Code to the latest version
+- Completely close VS Code and launch it from the terminal with `code .` (instead of opening VS Code from the launch
+  icon)
+
+More context about this issue on https://github.com/Shopify/vscode-ruby-lsp/issues/901.
+
+### Bundler issues
+
+If the extension successfully activated the Ruby environment, it may still fail when trying to compose the custom bundle
+to run the server gem. This could be a regular bundler issue, like not being able to satisfy dependencies due to a
+conflicting version requirement, or it could be a configuration issue.
+
+For example, if the project has its linter/formatter put in an optional `Gemfile` group and that group is excluded in
+the Bundler configuration, the Ruby LSP will not be able to see those gems.
+
+```ruby
+# Gemfile
+
+# ...
+
+# If Bundler is configured to exclude this group, the Ruby LSP will not be able to find `rubocop`
+group :optional_group do
+  gem "rubocop"
+end
+```
+
+If you experience Bundler related issues, double-check both your global and project-specific configuration to check if
```

Comment:
Could we list a few file locations/commands that may help users on this? I think `bundle config --help` should be a very good first step for them to make the first step.